tools/osv-linter/internal/pkgchecker/ecosystems.go

```diff
@@ -0,0 +1,340 @@
+package pkgchecker
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"regexp"
+	"slices"
+	"strings"
+
+	"github.com/ossf/osv-schema/linter/internal/faulttolerant"
+	"github.com/tidwall/gjson"
+)
+
+// Dispatcher for ecosystem-specific package existence checking.
+func ExistsInEcosystem(pkg string, ecosystem string) bool {
```

Comment:
is there any plan to check existence for other ecosystems as well?

Reply:
Yes, I'll be progressively building these out